packages/fetchai/connections/soef/connection.py

```diff
@@ -425,19 +433,46 @@ def _is_compatible_query(query: Query) -> bool:
         :return: bool
         """
         is_compatible = True
-        is_compatible = is_compatible and len(query.constraints) == 1
-        constraint_one = query.constraints[0]
-        is_compatible = is_compatible and isinstance(constraint_one, Constraint)
+        is_compatible = is_compatible and len(query.constraints) >= 1
+        constraint_distances = [
+            c
+            for c in query.constraints
+            if isinstance(c, Constraint)
+            and c.constraint_type.type == ConstraintTypes.DISTANCE
+        ]
+        is_compatible = is_compatible and len(constraint_distances) == 1
         if is_compatible:
-            constraint_one = cast(Constraint, constraint_one)
+            constraint_distance = cast(Constraint, constraint_distances[0])
             is_compatible = is_compatible and (
-                set([constraint_one.attribute_name]) == set(["location"])
-                and set([constraint_one.constraint_type.type])
+                set([constraint_distance.attribute_name]) == set(["location"])
+                and set([constraint_distance.constraint_type.type])
                 == set([ConstraintTypes.DISTANCE])
             )
         return is_compatible
 
-    def _find_around_me(self, radius: float) -> None:
+    @staticmethod
+    def _construct_personality_filter_params(
```

Comment:
If it is a static method doesn't mean that I have access from any script as soon as I import the connection ? I mean does this have to be private ?

Reply:
It could be public but we don't write these to be used outside the framework.